include/boost/random/exponential_distribution.hpp

```diff
@@ -23,11 +24,196 @@
+    0.38677382908413768115, 0.39093173698479710717, 0.39513698183329015336, 0.39939068447523107877,
+    0.40369401253053026739, 0.40804818315203238238, 0.41245446599716116772, 0.41691418643300289465,
+    0.42142872899761659635, 0.42599954114303435739, 0.43062813728845883923, 0.43531610321563659758,
+    0.44006510084235387501, 0.44487687341454851593, 0.44975325116275498919, 0.45469615747461548049,
+    0.45970761564213768669, 0.46478975625042618067, 0.46994482528395999841, 0.47517519303737738299,
+    0.48048336393045423016, 0.48587198734188493564, 0.49134386959403255500, 0.49690198724154955294,
+    0.50254950184134769289, 0.50828977641064283495, 0.51412639381474855788, 0.52006317736823356823,
+    0.52610421398361972602, 0.53225388026304326945, 0.53851687200286186590, 0.54489823767243963663,
+    0.55140341654064131685, 0.55803828226258748140, 0.56480919291240022434, 0.57172304866482579008,
+    0.57878735860284503057, 0.58601031847726802755, 0.59340090169173341521, 0.60096896636523224742,
+    0.60872538207962206507, 0.61668218091520762326, 0.62485273870366592605, 0.63325199421436607968,
+    0.64189671642726607018, 0.65080583341457104881, 0.66000084107899974178, 0.66950631673192477684,
+    0.67935057226476538741, 0.68956649611707798890, 0.70019265508278816709, 0.71127476080507597882,
+    0.72286765959357200702, 0.73503809243142351530, 0.74786862198519510742, 0.76146338884989624862,
+    0.77595685204011559675, 0.79152763697249565519, 0.80842165152300838005, 0.82699329664305033399,
+    0.84778550062398962096, 0.87170433238120363669, 0.90046992992574643800, 0.93814368086217467916,
+    1
+};
+
+template<class RealType = double>
+struct unit_exponential_distribution
+{
+    template<class Engine>
+    RealType operator()(Engine& eng) {
+        const double * const table_x = exponential_table<double>::table_x;
+        const double * const table_y = exponential_table<double>::table_y;
+        RealType shift(0);
+        for(;;) {
+            std::pair<RealType, int> vals = generate_int_float_pair<RealType, 8>(eng);
+            int i = vals.second;
+            RealType x = vals.first * RealType(table_x[i]);
+            if(x < table_x[i + 1]) return shift + x;
+            // For i=0 we need to generate from the tail, but because this is an exponential
+            // distribution, the tail looks exactly like the body, so we can simply repeat with a
+            // shift:
+            if (i == 0) shift += RealType(table_x[1]);
+            else {
+                RealType y = RealType(table_y[i]) + uniform_01<RealType>()(eng) * RealType(table_y[i + 1] - table_y[i]);
+                // We only need to bother checking f() for points below the diagonal; points above
+                // will be rejected (because f() is convex).
+                if ((table_x[i] - table_x[i + 1]) * (y - table_y[i]) < (x - table_x[i]) * (table_y[i] - table_y[i + 1])
```

Comment:
This probably makes no measurable difference, but does the compiler CSE y[i]-y[i+1] and y[i+1]-y[i] on the previous line?